diff --git a/node_modules/pouchdb-core/lib/index.js b/node_modules/pouchdb-core/lib/index.js
index 824f271..15e8c2c 100644
--- a/node_modules/pouchdb-core/lib/index.js
+++ b/node_modules/pouchdb-core/lib/index.js
@@ -890,7 +890,7 @@ AbstractPouchDB.prototype.getAttachment =
     }
     if (res.doc._attachments && res.doc._attachments[attachmentId]) {
       opts.ctx = res.ctx;
-      opts.binary = true;
+      opts.binary = false;
       self._getAttachment(docId, attachmentId,
                           res.doc._attachments[attachmentId], opts, callback);
     } else {
